backend/modules/dataloaders/truefoundryloader.py

```diff
@@ -23,98 +24,100 @@ def load_filtered_data(
-            datasource_type = "artifact"
-            # uri should be the artifcat fqn that includes the version number
-            artifact_fqn = data_source.uri
-            # Get information about the data directory and download it to the destination directory.
-            logger.info("Downloading Artifact: {}".format(artifact_fqn))
-
-            # Get the artifact version directly
-            dataset = truefoundry_client.get_artifact_version_by_fqn(artifact_fqn)
-            # download it to disk
-            # `artifact_files_dir` points to a directory that has all contents of the artifact
-            artifact_files_dir = dataset.download(path=dest_dir)
-            logger.debug(f"Artifact download info: {artifact_files_dir}")
-
-        elif data_source.uri.startswith("data-dir"):
-            datasource_type = "data-dir"
-            # Data source URI is the FQN of the data directory.
-            dataset = truefoundry_client.get_data_directory_by_fqn(data_source.uri)
-            artifact_files_dir = dataset.download(path=dest_dir)
-            logger.debug(f"Data directory download info: {artifact_files_dir}")
-
-        else:
-            raise ValueError(f"Unsupported data_source uri type {data_source.uri}")
-
-        if artifact_files_dir:
-            if os.path.exists(os.path.join(artifact_files_dir, "files")):
-                logger.debug("Files directory exists")
-                artifact_files_dir = os.path.join(artifact_files_dir, "files")
+        tfy_files_dir = None
+
+        # Check if the data source type is allowed
+        if not any(
+            data_source.type.startswith(allowed_type)
+            for allowed_type in ALLOWED_DATA_SOURCE_TYPES
+        ):
+            raise ValueError(f"Unsupported data source type {data_source.type}")
+
+        # If the data source type is data-dir, download the data directory from TrueFoundry.
+        if data_source.uri.startswith("data-dir"):
+            try:
+                # Data source URI contains the Truefoundry FQN(Fully Qualified Name) of the data directory.
+                # Use the FQN to get the data directory from TrueFoundry.
+                dataset = TRUEFOUNDRY_CLIENT.get_data_directory_by_fqn(data_source.uri)
+                # Download the data directory to the destination directory.
+                tfy_files_dir = dataset.download(path=dest_dir)
+                logger.debug(f"Data directory download info: {tfy_files_dir}")
+            except Exception as e:
+                logger.error(f"Error downloading data directory: {str(e)}")
+                raise ValueError(f"Failed to download data directory: {str(e)}")
+
+        # If tfy_files_dir is None, it means the data was not downloaded.
+        if tfy_files_dir is None:
+            logger.error("Download info not found")
+            return iter([])
+
+        # TODO: Why only limit to files directory? Can we not have other files in the data directory?
```

Comment:
All the files that are uploaded go within the files folder of the created data-dir.

Reply:
got it. will remove the comment

Reply:
removed in latest commits